lib/ansible/template/native_helpers.py

```diff
@@ -37,6 +38,17 @@ def ansible_native_concat(nodes):
         if isinstance(out, AnsibleVaultEncryptedUnicode):
             return out.data
 
+        if isinstance(out, StrictUndefined):
+            # A hack to raise proper UndefinedError/AnsibleUndefinedVariable exception.
+            # We need to access the AnsibleUndefined(StrictUndefined) object by either of the following:
+            # __iter__, __str__, __len__, __nonzero__, __eq__, __ne__, __bool__, __hash__
+            # to actually raise the exception.
+            # (see Jinja2 source of StrictUndefined to get up to date info)
+            # Otherwise the undefined error would be raised on the next access which might not be properly handled.
+            # See https://github.com/ansible/ansible/issues/52158
+            # We do that only here because it is taken care of by text_type() in the else block below already.
+            str(out)
```

Comment:
It was cool to see the form your solution took. It's super targeted, because it won't touch the code path if the setting is off, and is kind of defensive in how it uses type checking first.

Reply:
:+1: It was fun too.